conn.go

```diff
@@ -91,17 +92,18 @@ func (l *chanListener) Addr() net.Addr {
 
 func (l *chanListener) Close() error {
 	l.closeOnce.Do(func() {
-		close(l.conns)
+		close(l.closed)
```

Comment:
*question*: can we also close `l.conns` here ? 

Will the select statement in `mux.go` always choose the `l.closed` case because the channel is closed and always available ?

Reply:
we actually can, I don't think it will make any difference though

Reply:
but it will catch us placing stuff into conns when it is closed. I'll update the PR

Reply:
Is that safe for `handleGRPC`? If you send to a closed channel it'll panic. Not sure off hand what happens in a select.

Reply:
ha! I think @kruskall and I didn't have the right understanding of the select statement guarantees:
from https://go.dev/ref/spec#Select_statements
> If one or more of the communications can proceed, a single one that can proceed is chosen via a uniform pseudo-random selection. Otherwise, if there is a default case, that case is chosen. If there is no default case, the "select" statement blocks until at least one of the communications can proceed. 

So @axw is right, it will panic because select priority is not guaranteed. I'll revert the commit.

Reply:
Unfortunately it also means that we do not reliably return `errors.New("grpc listener closed")`.